### ISSUE TITLE
Undo / Redo through keyboard shortcuts not working properly

### ISSUE THREAD

**Issue by H16Y** [marktext/marktext#619]
### Description

This happens occasionally, but if you follow the steps below, there is a large probability to trigger the issue.
The undo / redo through menu is working properly. (Yet once the issue is triggered, it cannot be fixed with menu.)

- [x] Can you reproduce the issue? 

### Steps to reproduce

1. Create a title (h1).
2. Start a new line and write something.
3. Create a second title (h2).
4. Start several lines and write random long strings. (Better be long and the typing should be fast.)
5. Press `Ctrl+Z` at least 2 times.
6. You may observe that at least one time, `Ctrl+Z` has no effect on the content. Then the bug is triggered.
7. Press `Ctrl+Shift+Z`. At least one redo cannot be operated.

**Expected behavior:**

Precise Undo / Redo history.

**Actual behavior:**

(Part of) redo history is lost.

**Link to an example: [optional]**

![4](https://user-images.githubusercontent.com/45055956/50272766-bd963a00-0473-11e9-94a5-d8bbf4422ed7.gif)

### Versions

- Mark Text: v 0.13.65
- Operating system: Windows 10 1607 x64

**Comment by fxha** [comment]
Maybe a duplicate of #542? It seems to be an issue with our undo/redo stack and the 200ms text commit rate but we will rewrite it soon.

**Comment by H16Y** [comment]
@fxha Yes, I think so. Sorry that I did not notice that issue.
But in my experiment, I only pressed `Ctrl+Z` for two times, and I could not redo. But if I use Undo / Redo through menu, I do not experience that problem. Are there any differences between the code of menu and the code of keyboard shortcuts?

Besides, can you recommend some methods to enable us without Electron development environment to see the error messages? I tried using chrome to debug remotely, but I did not see any error messages. Thanks!

**Comment by fxha** [comment]
>  But if I use Undo / Redo through menu, I do not experience that problem. Are there any differences between the code of menu and the code of keyboard shortcuts?

No, both call the same function. You could start Mark Text with `--debug` and set `MARKTEXT_DEBUG_KEYBOARD` environment variable to `1` to see all received shortcuts in your terminal. E.g on Linux:

```
$ export MARKTEXT_DEBUG_KEYBOARD=1
$ cd into-mark-text-directory
$ ./marktext --debug
```

Then you will receive something like this:

```
You pressed CmdOrCtrl+Z
You pressed CmdOrCtrl+Shift+Z
```

> can you recommend some methods to enable us without Electron development environment to see the error messages

You can use `--debug` since `v0.13.65` and open dev. tools via `View -> Toggle dev. tools` but you will receive strange looking error messages. We fixed this issue but you have to wait until `v0.14.x` or build it from `master`.

**Comment by amundv** [comment]
This seems to still be a problem on macOS v.10.13.3 with Mark Text v.0.14.0
